solidity/contracts/liquidity-protection/LiquidityProtectionStore.sol

```diff
@@ -527,28 +545,120 @@ contract LiquidityProtectionStore is ILiquidityProtectionStore, Owned, TokenHand
         emit SystemBalanceUpdated(_token, prevAmount, newAmount);
     }
 
-    /**
-     * @dev returns the total protected pool token amount for a given pool
-     *
-     * @param _poolToken   pool token address
-     * @return total protected amount
-     */
-    function totalProtectedPoolAmount(IDSToken _poolToken) external view returns (uint256) {
-        return totalProtectedPoolAmounts[_poolToken];
+    function toUint128(uint256 _amount) private pure returns (uint128) {
+        require(_amount <= MAX_UINT128, "ERR_AMOUNT_TOO_HIGH");
+        return uint128(_amount);
     }
 
-    /**
-     * @dev returns the total protected reserve amount for a given pool
-     *
-     * @param _poolToken       pool token address
-     * @param _reserveToken    reserve token address
-     * @return total protected amount
-     */
-    function totalProtectedReserveAmount(IDSToken _poolToken, IERC20Token _reserveToken)
-        external
-        view
-        returns (uint256)
-    {
-        return totalProtectedReserveAmounts[_poolToken][_reserveToken];
+    function encodeReserveRateInfo(
+        uint256 _reserveRateN,
+        uint256 _reserveRateD,
+        uint256 _reserveRateT
+    ) private pure returns (uint256) {
+        assert(_reserveRateN <= MAX_UINT112 && _reserveRateD <= MAX_UINT112 && _reserveRateT <= MAX_UINT32);
+        return _reserveRateN | (_reserveRateD << 112) | (_reserveRateT << 224);
+    }
+
+    function decodeReserveRateN(uint256 _reserveRateInfo) private pure returns (uint256) {
+        return _reserveRateInfo & MAX_UINT112;
+    }
+
+    function decodeReserveRateD(uint256 _reserveRateInfo) private pure returns (uint256) {
+        return (_reserveRateInfo >> 112) & MAX_UINT112;
+    }
+
+    function decodeReserveRateT(uint256 _reserveRateInfo) private pure returns (uint256) {
+        return _reserveRateInfo >> 224;
+    }
+
+    function seedNextPositionId(uint256 _nextPositionId) external seederOnly {
+        nextPositionId = _nextPositionId;
+    }
+
+    function seedPositions(
```

Comment:
it's a monster! :D
technically we could have used a separate contract for migration to keep this one cleaner but it will increase gas costs.
in this specific case, since it will probably only increase gas costs by roughly 1-2%, we should consider it

Reply:
Another contract which would be the seeder of this contract?
That would increase the gas cost by a lot more than 1-2% IMO, plus, it won't even make that "monster" any less of a monster than it already is. Instead of input arrays, we will have input variables, and the `for` loop will be removed. That's very little reduction in text size, plus, why do we care about this text in the contract anyway?
On top of all of that, adding another mediator here (the seeding contract) will just add a lot more complexity to the already-complicated migration process.